lib/pharos/phase_manager.rb

```diff
@@ -19,25 +19,37 @@ def initialize(ssh_manager:, **options)
 
     # @param phases [Array<Pharos::Phases::Base>]
     # @return [Array<...>]
-    def run_parallel(phases)
+    def run_parallel(phases, &block)
       threads = phases.map { |phase|
         Thread.new do
-          begin
-            yield phase
-          rescue StandardError => exc
-            puts " [#{phase}] #{exc.class}: #{exc.message}"
-            raise
-          end
+          yield_phase_with_retry(phase, &block)
         end
       }
       threads.map(&:value)
     end
 
     # @param phases [Array<Pharos::Phases::Base>]
     # @return [Array<...>]
-    def run_serial(phases)
+    def run_serial(phases, &block)
       phases.map do |phase|
+        yield_phase_with_retry(phase, &block)
+      end
+    end
+
+    # @param phase [Pharos::Phases::Base]
+    # @param retry_times [Integer]
+    def yield_phase_with_retry(phase, retry_times = 10)
+      retries = 0
+      begin
         yield phase
+      rescue StandardError => exc
```

Comment:
`StandardError` also includes things like ruby `NameError`... I don't think it's a good idea to retry all phases on any error like that, there should be an explicit per-phase list of errors to retry on... something like

```ruby
class ConfigureWeave < Pharos::Phase
    retry_error K8s::Error
    
    ...
end
```

Reply:
Idk if explicit list per-phase is a good idea.. because then almost every phase needs to list OpenSSL etc errors?

Reply:
Maybe fixed list of more specific errors?

Reply:
Now it retries only on specific errors (kube, openssl, ssh).